### PR TITLE
Changes for v2.2

### PR DESCRIPTION
**Pull request by kartographer**
Updating pyuvdata version to v2.2.0

## Description

Several methods have been removed that were slated deprecation, specifically:
- The UVBeam methods `set_cs_params`, `set_efield`, `set_power`, `set_simple`, and
`set_phased_array` have been removed.
- The UVCal methods `set_gain`, `set_delay`, `set_unknown_cal_type`, `set_sky`, and
`set_redundant` have been removed.
- The UVData methods `set_phased` and `set_drift` have been removed.
- The functions `utils._str_to_bytes` and `utils._bytes_to_str` have been removed.
- The `flag_missing` keyword of the `utils.uvcalibrate` function has been removed.

In addition to the above, the `flag_missing` keyword of `utils.uvcalibrate` has been removed, and the change log updated.

## Motivation and Context

Fast-follow version update following the changes in #979.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [x] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Breaking change checklist:
- [x] I have updated the docstrings associated with my change using the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] I have updated the tutorial to reflect my changes (if appropriate).
- [ ] My change includes backwards compatibility and deprecation warnings (if possible).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests pass.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/CHANGELOG.md).

Version change checklist:
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/CHANGELOG.md) to put all the unreleased changes under the new version (leaving the unreleased section empty).
- [x] I have noted any dependency changes since the last version and will update the conda package build accordingly.

### CODE REVIEW

**Comment by kartographer** [comment]
This could probably use a lookover from someone skilled w/ UVCal -- in particular, the `utils.uvcalibrate` function and the `test_*_oldfiles` tests in `test_utils`.

File: docs/tutorial.rst [hunk 1]
```diff
@@ -1684,10 +1684,10 @@ a) Calibration of UVData by UVCal
   >>> UV.read(os.path.join(DATA_PATH, 'zen.2458116.30448.HH.uvh5'))
   >>> UVC = UVCal()
   >>> UVC.read_calfits(os.path.join(DATA_PATH, 'zen.2458116.30448.HH.flagged_abs.calfits'))
-  >>> UV_calibrated = utils.uvcalibrate(UV, UVC, inplace=False)
+  >>> UV_calibrated = utils.uvcalibrate(UV, UVC, inplace=False, ant_check=False)
 
   >>> # We can also un-calibrate using the same UVCal
-  >>> UV_uncalibrated = utils.uvcalibrate(UV_calibrated, UVC, inplace=False, undo=True)
+  >>> UV_uncalibrated = utils.uvcalibrate(UV_calibrated, UVC, inplace=False, undo=True, ant_check=False)
```
Comment (bhazelton) [inline docs/tutorial.rst:1690]: I think we want to use different files that don't require setting `ant_check=False`. That's not something we really want to encourage people to do. I can make that change.

**Review by bhazelton — COMMENTED** [review]
Looks pretty good, but a few comments about uvcalibrate that I can take care of.

**Review by bhazelton — APPROVED** [review]
Looks good now!

**Pull request merged**